src/components/ChangePlan/Checkout/BillingInformation/BillingInformation.js

```diff
@@ -0,0 +1,305 @@
+import React, { useEffect, useState } from 'react';
+import { InjectAppServices } from '../../../../services/pure-di';
+import { useIntl } from 'react-intl';
+import useTimeout from '../../../../hooks/useTimeout';
+import {
+  FieldGroup,
+  FieldItem,
+  InputFieldItem,
+  PhoneFieldItem,
+  SelectFieldItem,
+  CheckboxFieldItem,
+  SubmitButton,
+} from '../../../form-helpers/form-helpers';
+import { Form, Formik } from 'formik';
+import { getCountries, getFormInitialValues } from '../../../../utils';
+import { Loading } from '../../../Loading/Loading';
+
+const fieldNames = {
+  sameAddressAsContact: 'sameAddressAsContact',
+  firstname: 'firstname',
+  lastname: 'lastname',
+  address: 'address',
+  city: 'city',
+  province: 'province',
+  country: 'country',
+  zipCode: 'zipCode',
+  phone: 'phone',
+};
+
+const getBillingInformation = (contactInformation) => {
+  return {
+    firstname: contactInformation.firstname,
+    lastname: contactInformation.lastname,
+    address: contactInformation.address,
+    city: contactInformation.city,
+    province: contactInformation.province,
+    country: contactInformation.country,
+    zipCode: contactInformation.zipCode,
+    phone: contactInformation.phone,
+    sameAddressAsContact: true,
+  };
+};
+
+export const BillingInformation = InjectAppServices(
+  ({
+    dependencies: { dopplerUserApiClient, dopplerBillingUserApiClient, staticDataClient },
+    handleSaveAndContinue,
+    showTitle,
+  }) => {
+    const [state, setState] = useState({ loading: true });
+    const [formSubmitted, setFormSubmitted] = useState(false);
+    const [states, setStates] = useState([]);
+    const [sameAddressInformation, setSameAddressInformation] = useState(false);
+    const createTimeout = useTimeout();
+    const intl = useIntl();
+
+    useEffect(() => {
+      const fetchData = async () => {
+        const contactInformationResult = await dopplerUserApiClient.getContactInformationData();
+        const billingInformationResult =
+          await dopplerBillingUserApiClient.getBillingInformationData();
+        const statesResult = await staticDataClient.getStatesData(
+          billingInformationResult.success
```

Comment:
nested ternary operators make readability very difficult. Maybe you can do it another way. 
simple checks should be used.

Reply:
In my opinion, if you use the right order and indentation it is very clear:

```
        const statesResult = await staticDataClient.getStatesData(
          !billingInformationResult.success ? 'ar'
          : billingInformationResult.value.country !== '' ? billingInformationResult.value.country
          : contactInformationResult.value.country;
```

Reply:
I removed the nested ternary operators